README.md

```diff
@@ -1,61 +1,2367 @@
+```
+COPY package.json index.js /usr/src/app/
+
+RUN npm install
+```
+to:
+```
+COPY package.json /usr/src/app/
+
+RUN npm install
+
+COPY index.js /usr/src/app/
+```
+And rebuild the image again:
+```
+$ docker build -t simple-docker-app:latest .
+Sending build context to Docker daemon   5.12kB
+Step 1/7 : FROM balenalib/intel-nuc-debian-node:latest
+ ---> 4498fe12b99e
+Step 2/7 : EXPOSE 9854
+ ---> Using cache
+ ---> ff63da014498
+Step 3/7 : WORKDIR /usr/src/app
+ ---> Using cache
+ ---> 568a5e43a29e
+Step 4/7 : COPY package.json /usr/src/app/
+ ---> 2c51ccf2a7b2
+Step 5/7 : RUN npm install
+ ---> Running in 02c04604c1f0
+npm notice created a lockfile as package-lock.json. You should commit this file.
+npm WARN simple-docker-app@1.0.0 No description
+npm WARN simple-docker-app@1.0.0 No repository field.
+
+added 50 packages from 37 contributors and audited 126 packages in 2.638s
+found 0 vulnerabilities
+
+Removing intermediate container 02c04604c1f0
+ ---> 1705ac26c046
+Step 6/7 : COPY index.js /usr/src/app/
+ ---> eb4a5841b1f5
+Step 7/7 : CMD ["/usr/local/bin/npm", "start"]
+ ---> Running in faf3d745a235
+Removing intermediate container faf3d745a235
+ ---> 3a8c0a432bc2
+Successfully built 3a8c0a432bc2
+Successfully tagged simple-docker-app:latest
+```
+Now change the `index.js` source file again and re-add those two comment lines:
+```
+// This comment changes the image
+// Another comment changes the image
+```
+at the start of the file. Finally let's rebuild again:
+```
+$ docker build -t simple-docker-app:latest .
+Sending build context to Docker daemon   5.12kB
+Step 1/7 : FROM balenalib/intel-nuc-debian-node:latest
+ ---> 4498fe12b99e
+Step 2/7 : EXPOSE 9854
+ ---> Using cache
+ ---> ff63da014498
+Step 3/7 : WORKDIR /usr/src/app
+ ---> Using cache
+ ---> 568a5e43a29e
+Step 4/7 : COPY package.json /usr/src/app/
+ ---> Using cache
+ ---> 2c51ccf2a7b2
+Step 5/7 : RUN npm install
+ ---> Using cache
+ ---> 1705ac26c046
+Step 6/7 : COPY index.js /usr/src/app/
+ ---> 5233ceadf248
+Step 7/7 : CMD ["/usr/local/bin/npm", "start"]
+ ---> Running in 0c15ac7cfc32
+Removing intermediate container 0c15ac7cfc32
+ ---> 81a2fcf7791c
+Successfully built 81a2fcf7791c
+Successfully tagged simple-docker-app:latest
+```
+The build was far quicker as NPM didn't have to reinstall the used modules
+again, and the filesystem layer already built was reused:
+```
+Step 5/7 : RUN npm install
+ ---> Using cache
+```
+By considering how your Dockerfile is constructed, and how the cache may be
+utilized is an extremely valuable part of using Docker. Whilst the example
+we've used is very simplistic, for builds where a Dockerfile is many lines long
+this can be critical in cutting down build time from hours to minutes.
+
+### 8. Running Containers in Detached, Command and Interactive Modes
+
+So far we've `docker run` to create a new container, which outputs all STDIO
+to the console.
+
+However, there are multiple ways in which a container can be instanced.
```

Comment:
I'm not familiar with "instanced" in this context, is that common terminology? Should it be instantiated or just created to simplify.

Reply:
Instanced is used, but I'll change it to instantiated. They're both over docker docs, so it's worth a reader getting used to them.